From 9b707dbcf59f74099cfd2afaa25ae01aeb624323 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:49:45 -0400
Subject: [tools] lib/traceevent: Remove malloc_or_die from event-plugin.c

Message-id: <1408953210-26343-98-git-send-email-jolsa@redhat.com>
Patchwork-id: 88801
O-Subject: [PATCH RHEL7.1 BZ1133083 097/322] tools lib traceevent: Remove malloc_or_die from event-plugin.c
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit f9bb36afb25d3bfda1c9276a55985b710c8a91ae
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Dec 3 14:09:36 2013 +0100

description
===========
Removing malloc_or_die calls from event-plugin.c,
replacing them with standard malloc and error path.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/lib/traceevent/event-plugin.c b/tools/lib/traceevent/event-plugin.c
index d272d87..125f567 100644
--- a/tools/lib/traceevent/event-plugin.c
+++ b/tools/lib/traceevent/event-plugin.c
@@ -47,7 +47,11 @@ load_plugin(struct pevent *pevent, const char *path,
  char *plugin;
  void *handle;
 
- plugin = malloc_or_die(strlen(path) + strlen(file) + 2);
+ plugin = malloc(strlen(path) + strlen(file) + 2);
+ if (!plugin) {
+  warning("could not allocate plugin memory\n");
+  return;
+ }
 
  strcpy(plugin, path);
  strcat(plugin, "/");
@@ -71,7 +75,12 @@ load_plugin(struct pevent *pevent, const char *path,
   goto out_free;
  }
 
- list = malloc_or_die(sizeof(*list));
+ list = malloc(sizeof(*list));
+ if (!list) {
+  warning("could not allocate plugin memory\n");
+  goto out_free;
+ }
+
  list->next = *plugin_list;
  list->handle = handle;
  list->name = plugin;
@@ -163,7 +172,11 @@ load_plugins(struct pevent *pevent, const char *suffix,
  if (!home)
   return;
 
- path = malloc_or_die(strlen(home) + strlen(LOCAL_PLUGIN_DIR) + 2);
+ path = malloc(strlen(home) + strlen(LOCAL_PLUGIN_DIR) + 2);
+ if (!path) {
+  warning("could not allocate plugin memory\n");
+  return;
+ }
 
  strcpy(path, home);
  strcat(path, "/");
-- 
1.7.1